torch/autograd/graph.py

```diff
@@ -821,10 +821,16 @@ def _engine_run_backward(
     attach_logging_hooks = log.getEffectiveLevel() <= logging.DEBUG
     if attach_logging_hooks:
         unregister_hooks = _register_logging_hooks_on_whole_graph(t_outputs)
+    compiled_autograd_ctx = (
+        torch._dynamo.compiled_autograd.disable()
+        if torch._dynamo.compiled_autograd.in_compiled_autograd_region
+        else contextlib.nullcontext()
+    )
     try:
-        return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
-            t_outputs, *args, **kwargs
-        )  # Calls into the C++ engine to run the backward pass
+        with compiled_autograd_ctx:
```

Comment:
could we move the context creation logic into some helper in torch/_dynamo/compiled_autograd.py? and keep minimal logic here

```suggestion
        with torch._dynamo.compiled_autograd.maybe_disable_compiled_autograd():
```

Reply:
done